### ISSUE TITLE
List of new features since SciPy 1.0

### ISSUE THREAD

**Issue by j-bowhay** [scipy/scipy-articles#274]
I have gone through all of the release notes since SciPy 1.0 to collate at list of features added. Note, I don't imagine everything in this list will get a mention in the paper but hopefully it will be a helpful starting point.

Please feel free to edit if I have missed anything. Likewise feel group related functionality together and improve any formatting - I was on the train while doing this so didn't have the patience to make this look too beautiful!

## SciPy Wide Improvements / Features

- Array API
- Fortran free
- Elementwise-iterative methods (`nsum`, `tanhsinh`, `scipy.differentiate`, `scipy.optimize.elementwise`)
- free-threaded Python support
- SPEC 7
- meson
- separate scipy-release repo (security)
- github actions (move off appveyor/travis)
- trusted publishing to PyPI (security)
- new platforms: manylinux-aarch64, win-arm64, macos-arm64, musllinux-x86_64, etc. We've come up with rationalisation for tiered support.
- Developed AI policy.
- ? when did nightly wheels come in ?
- scipy_openblas32/64 wheels
- policy for bringing new people onto the team
- number of new maintainers
- other governance, such as CoC policy
- shrunk size of binaries

## `scipy.cluster`
- `DisjointSet`
## `scipy.constants`
- CODATA 2018
## `scipy.datasets`
- New module for datasets
## `scipy.differentiate`
- New module for differentiation!
## `scipy.fft`
- New module superseeding `fftpack`
## `scipy.fftpack`
- Legacy
## `scipy.integrate`
- `cubature`
- `lebedev_rule`
- `cumulative_simpson `
- `qmc_quad `
- `quad_vec`
## `scipy.interpolate`
- `AAA`
- `FloaterHormannInterpolator`
- `make_splrep` / `make_splprep` / `generate_knots` / `make_smoothing_spline`
- Modified Akima Interpolation
- `NdBSpline`
## `scipy.io`
- No significant new features
## `scipy.linalg`
- ILP64 support
- Batched input support
- More LAPACK wrappers
- structure detection for `solve` etc
- Accelerate support
- Structure detection: `bandwidth`, `issymmetric`, `ishermitian`
- `matmul_toeplitz`
- `cossin`
- `khatri_rao`
- `convolution_matrix`
- `fiedler` / `fiedler_companion`
- `ldl`
- `null_space`
- `cdf2rdf`
- `rsf2csf`
## `scipy.ndimage`
- `vectorized_filter`
- `value_indices`
## `scipy.odr`
- Removed
## `scipy.optimize`
- Migration of COBYLA to PRIMA
- `cobyqa`
- `isotonic_regression`
- HiGHS (`milp`)
- `direct`
- `quadratic_assignment`
- `toms748`
- `root_scalar`
- `dual_annealing`
- `shgo`
- `trust-constr`
- parallelization of Jacobian/Hessian calculation, `differential_evolution` using the `workers` keyword.
- `differential_evolution` uses constraints
## `scipy.signal`
- Whittaker-Henderson smoothing
- `firwin_2d`
- `closest_STFT_dual_window`
- `ShortTimeFFT`
- `envelope`
- New windows:`lanczos`, `kaiser_bessel_derived`, `taylor`, `dpss`, `general_cosine` and `general_hamming`.
- `CZT`
- `ZoomFFT`
- `gammatone`
- `iircomb`
- `correlation_lags`
- `oaconvolve`
- `find_peaks` / `peak_prominences` / `peak_widths`
- ZPK versions of frequency transformations: scipy.signal.bilinear_zpk, scipy.signal.lp2bp_zpk, scipy.signal.lp2bs_zpk, scipy.signal.lp2hp_zpk, scipy.signal.lp2lp_zpk.
## `scipy.sparse`
- New sparse array interface
- nD `coo` arrays
- n-dimensional linear operators added to scipy.sparse.linalg.LinearOperator
- `funm_multiply_krylov` a restarted Krylov method for evaluating `y = f(tA) b`
- `csgraph.yen`
- `pydata/sparse` compatibility
- `linalg.matrix_power`
- PROPACK
- `tfqmr`
- `min_weight_full_bipartite_matching`
## `scipy.spatial`
- `transform.RigidTransform`
- `geometric_slerp`
- `distance.jensenshannon`
## `scipy.special`
- Improved accuracy of many functions by wrapping Boost
- xsf
- improved Legendre function implementations with a more consistent interface
- New special functions: `log_wright_bessel`, `stirling2`, `betaincc`, `betainccinv`, `powm1`, `log_expit`, `ellipr{c,d,f,g,j}`, `wright_bessel`, `ndtri_exp`, `log_softmax`, `voigt_profile`, `softmax`, `owens_t.`
## `scipy.stats`
- New distribution infrastructure
- axis-nan policy
- `qmc` submodule
- `sampling` submodule / UNU.RAN
- New distributions: matrix t, logistic distributions, dpareto_lognorm, landau, normal_inverse_gamma, poisson_binom, jf_skew_t, vonmises_fisher, rel_breitwigner, dirichlet_multinomial, uniform_direction, random_table, truncpareto, skewcauchy, zipfian, nchypergeom_fisher, nchypergeom_wallenius, genhyperbolic, studentized_range, laplace_asymmetric, nhypergeom, multivariate_t, multivariate_hypergeom, geninvgauss, betabinom, yulesimon, moyal, norminvgauss
- `lmoment`
- `chatterjeexi`
- `power`
- `quantile_test`
- `bws_test`
- `sobol_indices`
- `dunnett`
- `ecdf`
- `logrank`
- `false_discovery_control`
- `CensoredData`
- scipy.stats.MonteCarloMethod, scipy.stats.PermutationMethod, and scipy.stats.BootstrapMethod
- `goodness_of_fit`
- `poisson_means_test`
- `odds_ratio`
- `directional_stats`
- `expectile`
- `fit`
- `monte_carlo_test`
- `gzscore`
- `tukey_hsd`
- `binomtest`
- `cramervonmises_2samp`
- `alexandergovern`
- `barnard_exact `
- `page_trend_test`
- `somersd`
- `differential_entropy`
- `relative_risk`
- `contingency.association`
- `cramervonmises`
- `multiscale_graphcorr`
- `epps_singleton_2samp`
- `median_absolute_deviation`
- `gstd`
- `siegelslopes`

**Comment by chrisb83** [comment]
I created #279 with further information of the changes to the stats module. I hope it helps to get an overview of all the enhancements that we want to describe in the paper. As mentioned in #275 there are many changes.
